polaris-core/src/main/java/org/apache/polaris/core/persistence/transactional/TreeMapTransactionalPersistenceImpl.java

```diff
@@ -361,13 +362,19 @@ public List<EntityNameLookupRecord> lookupEntityActiveBatchInCurrentTxn(
             .map(
                 nameRecord ->
                     this.lookupEntityInCurrentTxn(
-                        callCtx, catalogId, nameRecord.getId(), entityType.getCode()))
-            .filter(entityFilter);
-    if (pageToken instanceof HasPageSize) {
-      data = data.limit(((HasPageSize) pageToken).getPageSize());
+                        callCtx, catalogId, nameRecord.getId(), entityType.getCode()));
+
+    Predicate<PolarisBaseEntity> tokenFilter = e -> true;
+    if (pageToken.hasDataReference()) {
```

Comment:
In cases like this, it's better to do a single assignment to prevent anything from breaking if lines get moved around:
```
final Predicate<PolarisBaseEntity> tokenFilter;
if (pageToken.hasDataReference()) {
  tokenFilter = e -> e.getId() > nextId;
} else {
  tokenFilter = e -> true;
}
```

Reply:
refactored as suggested :+1: